pkg/proxy/service/cluster_service_test.go

```diff
@@ -107,320 +107,372 @@ func (s *TestClusterServiceSuite) TestGetClusterAccess() {
-				require.NoError(s.T(), err)
-				require.NotNil(s.T(), ca)
-				expectedURL, err := url.Parse("https://api.endpoint.member-2.com:6443")
-				require.NoError(s.T(), err)
-				s.assertClusterAccess(access.NewClusterAccess(*expectedURL, expectedToken, "smith"), ca)
-				assert.Equal(s.T(), "smith2", ca.Username())
+				s.Run("verify cluster access no route", func() {
+					memberClient.MockGet = nil
+					expectedToken := "abc123" // should match member 2 bearer token
 
-				s.Run("another workspace on another cluster", func() {
 					// when
-					ca, err := svc.GetClusterAccess("789-ready", "", "teamspace", "") // workspace-context specified
+					ca, err := svc.GetClusterAccess("789-ready", "", "", "", publicViewerEnabled)
 
 					// then
 					require.NoError(s.T(), err)
 					require.NotNil(s.T(), ca)
-					expectedURL, err := url.Parse("https://api.endpoint.member-1.com:6443")
+					expectedURL, err := url.Parse("https://api.endpoint.member-2.com:6443")
 					require.NoError(s.T(), err)
-					s.assertClusterAccess(access.NewClusterAccess(*expectedURL, "def456", "smith"), ca)
 					assert.Equal(s.T(), "smith2", ca.Username())
+
+					s.assertClusterAccess(access.NewClusterAccess(*expectedURL, expectedToken, ""), ca)
+
+					s.Run("cluster access correct when username provided", func() {
+						// when
+						ca, err := svc.GetClusterAccess("", "smith@", "", "", publicViewerEnabled)
+
+						// then
+						require.NoError(s.T(), err)
+						require.NotNil(s.T(), ca)
+						expectedURL, err := url.Parse("https://api.endpoint.member-2.com:6443")
+						require.NoError(s.T(), err)
+						s.assertClusterAccess(access.NewClusterAccess(*expectedURL, expectedToken, "smith"), ca)
+						assert.Equal(s.T(), "smith2", ca.Username())
+					})
+
+					s.Run("cluster access correct when using workspace context", func() {
+						// when
+						ca, err := svc.GetClusterAccess("789-ready", "", "smith2", "", publicViewerEnabled) // workspace-context specified
+
+						// then
+						require.NoError(s.T(), err)
+						require.NotNil(s.T(), ca)
+						expectedURL, err := url.Parse("https://api.endpoint.member-2.com:6443")
+						require.NoError(s.T(), err)
+						s.assertClusterAccess(access.NewClusterAccess(*expectedURL, expectedToken, "smith"), ca)
+						assert.Equal(s.T(), "smith2", ca.Username())
+
+						s.Run("another workspace on another cluster", func() {
+							// when
+							ca, err := svc.GetClusterAccess("789-ready", "", "teamspace", "", publicViewerEnabled) // workspace-context specified
+
+							// then
+							require.NoError(s.T(), err)
+							require.NotNil(s.T(), ca)
+							expectedURL, err := url.Parse("https://api.endpoint.member-1.com:6443")
+							require.NoError(s.T(), err)
+							s.assertClusterAccess(access.NewClusterAccess(*expectedURL, "def456", "smith"), ca)
+							assert.Equal(s.T(), "smith2", ca.Username())
+						})
+					})
 				})
 			})
 		})
+	}
+
+	// public-viewer enabled
+	s.Run("user is public-viewer", func() {
```

Comment:
should we have test cases also with these workspaces:
- 456-not-ready
- 012-ready-unknown-cluster 

WDYT?